### PR TITLE
Allow computing the correlation matrix from the covariance matrix only

### PR DESCRIPTION
**Pull request by ASaragga**
Following issue #652 in StatsBase.jl

### CODE REVIEW

File: src/cov.jl [hunk 1]
```diff
@@ -131,6 +131,13 @@ deviations `s`. Use `StatsBase.cov2cor!` for an in-place version.
 """
 cov2cor(C::AbstractMatrix, s::AbstractArray) = cov2cor!(copy(C), s)
 
+"""
+     cov2cor(C)
+
+ Compute the correlation matrix from the covariance matrix `C`. Use `StatsBase.cov2cor!` for an in-place version.
+ """
```
Comment (nalimilan) [inline src/cov.jl:138]: I'd merge this with the docstring for the 2-argument version (with `cov2cor(C[, s])` as a signature).

**Comment by nalimilan** [comment]
Can you add tests?

**Comment by nalimilan** [comment]
Did you remove the tests on purpose?

**Comment by ASaragga** [comment]
Well, which tests? I didn’t add any tests to Statistics.jl yet. There where two tests that I added to StatsBase.jl in my PR, that failed for what are now obvious reasons.

**Comment by JockLawrie** [comment]
Hi there,

Looks like the method hasn't been implemented.
I can't edit this PR. Can someone add the following code? (and is it sufficient?)
Alternatively should I submit a separate PR?

In `src/cov.jl`
```julia
cov2cor!(C) = cov2cor!(C, [sqrt(C[i,i]) for i = 1:size(C, 1)])
cov2cor(C) = cov2cor!(copy(C))
```

In `test/cov.jl`
```julia
# Around line 126
@test cov2cor(cov1) ≈ cor1
@test cov2cor(cov2) ≈ cor2

# Around line 202
@test cov2cor(cov1) ≈ cor1
@test cov2cor(cov2) ≈ cor2

# Around line 209
StatsBase.cov2cor!(tmp_cov1)
@test tmp_cov1 ≈ cor1

# Around line 214
StatsBase.cov2cor!(tmp_cov2)
@test tmp_cov2 ≈ cor2
```

**Comment by nalimilan** [comment]
Please make a new PR, that's simpler.

**Comment by andreasnoack** [comment]
Superseded by https://github.com/JuliaStats/StatsBase.jl/pull/816

**Pull request closed without merging**